src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/button_press_request_test.cc

```diff
@@ -0,0 +1,297 @@
+    ON_CALL(mock_hmi_capabilities_, rc_capability())
+            .WillByDefault(Return(&rc_capabilities_));
+  }
+
+  MessageSharedPtr CreateBasicMessage() {
+    MessageSharedPtr message = CreateMessage();
+    (*message)[application_manager::strings::params]
+            [application_manager::strings::function_id] =
+        mobile_apis::FunctionID::ButtonPressID;
+    (*message)[application_manager::strings::params]
+            [application_manager::strings::connection_key] = kConnectionKey;
+    (*message)[application_manager::strings::params]
+            [application_manager::strings::connection_key] = kAppId;
+    return message;
+  }
+
+  template <class Command>
+  application_manager::SharedPtr<Command> CreateRCCommand(MessageSharedPtr& msg) {
+    InitCommand(kDefaultTimeout_);
+    return ::utils::MakeShared<Command>(mock_allocation_manager_,
+                                        msg ? msg : msg = CreateMessage(),
+                                        app_mngr_,
+                                        mock_rpc_service_,
+                                        mock_hmi_capabilities_,
+                                        mock_policy_handler_);
+  }
+
+ protected:
+  smart_objects::SmartObject rc_capabilities_;
+  utils::SharedPtr<MockApplication> mock_app_;
+  utils::SharedPtr<rc_rpc_plugin::RCAppExtension> rc_app_extention_;
+  test::components::policy_test::MockPolicyHandlerInterface
+      mock_policy_handler_;
+  testing::NiceMock<rc_rpc_plugin_test::MockResourceAllocationManager>
+      mock_allocation_manager_;
+
+};
+
+TEST_F(ButtonPressRequestTest,
+       Execute_ButtonNameMatchesModuleType_ExpectCorrectMessageSentToHMI) {
+  // Arrange
+  MessageSharedPtr mobile_message = CreateBasicMessage();
+  NsSmartDeviceLink::NsSmartObjects::SmartObject& msg_params = (*mobile_message)[application_manager::strings::msg_params];
+  msg_params[message_params::kModuleType] = "CLIMATE";
+  msg_params[message_params::kButtonName] = "AC";
+  msg_params[message_params::kButtonPressMode] = "SHORT";
+
+  // Expectations
+  const std::string resource = "CLIMATE";
+  ON_CALL(mock_policy_handler_, CheckModule(_,_))
+          .WillByDefault(Return(true));
+  EXPECT_CALL(mock_allocation_manager_, IsResourceFree(resource))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_allocation_manager_, AcquireResource(resource, _))
+      .WillOnce(Return(rc_rpc_plugin::AcquireResult::ALLOWED));
+  EXPECT_CALL(
+      mock_allocation_manager_,
+      SetResourceState(resource, kAppId, rc_rpc_plugin::ResourceState::BUSY));
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(HMIResultCodeIs(
+                                                        hmi_apis::FunctionID::Buttons_ButtonPress)))
+          .WillOnce(Return(true));
+
+  // Act
+  application_manager::SharedPtr<rc_rpc_plugin::commands::ButtonPressRequest> command =
+      CreateRCCommand<rc_rpc_plugin::commands::ButtonPressRequest>(mobile_message);
+  command->Run();
```

Comment:
@LitvinenkoIra Need to add `ASSERT_TRUE(command->Init());`  before `command->Run();`